src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -691,60 +683,68 @@ protected override bool TryComputeWeightedDistance(SyntaxNode leftNode, SyntaxNo
             }
         }
 
-        private static double ComputeWeightedDistanceOfLocalFunctions(LocalFunctionStatementSyntax leftNode, LocalFunctionStatementSyntax rightNode)
+        private static double ComputeWeightedDistanceOfNestedFunctions(SyntaxNode leftNode, SyntaxNode rightNode)
         {
-            double modifierDistance = ComputeDistance(leftNode.Modifiers, rightNode.Modifiers);
-            double returnTypeDistance = ComputeDistance(leftNode.ReturnType, rightNode.ReturnType);
-            double identifierDistance = ComputeDistance(leftNode.Identifier, rightNode.Identifier);
-            double typeParameterDistance = ComputeDistance(leftNode.TypeParameterList, rightNode.TypeParameterList);
-            double parameterDistance = ComputeDistance(leftNode.ParameterList.Parameters, rightNode.ParameterList.Parameters);
-            double bodyDistance = ComputeDistance((SyntaxNode)leftNode.Body ?? leftNode.ExpressionBody, (SyntaxNode)rightNode.Body ?? rightNode.ExpressionBody);
-
-            return
-                modifierDistance * 0.1 +
-                returnTypeDistance * 0.1 +
-                identifierDistance * 0.2 +
-                typeParameterDistance * 0.2 +
-                parameterDistance * 0.2 +
-                bodyDistance * 0.2;
-        }
-
-        private static double ComputeWeightedDistanceOfLambdas(SyntaxNode leftNode, SyntaxNode rightNode)
-        {
-            GetLambdaParts(leftNode, out var leftParameters, out var leftAsync, out var leftBody);
-            GetLambdaParts(rightNode, out var rightParameters, out var rightAsync, out var rightBody);
+            GetNestedFunctionsParts(leftNode, out var leftParameters, out var leftAsync, out var leftBody, out var leftModifiers, out var leftReturnType, out var leftIdentifier, out var leftTypeParameters);
+            GetNestedFunctionsParts(rightNode, out var rightParameters, out var rightAsync, out var rightBody, out var rightModifiers, out var rightReturnType, out var rightIdentifier, out var rightTypeParameters);
 
             if ((leftAsync.Kind() == SyntaxKind.AsyncKeyword) != (rightAsync.Kind() == SyntaxKind.AsyncKeyword))
             {
                 return 1.0;
             }
```

Comment:
We might want to reconsider this check in future
Could you add a comment with a ling to [https://github.com/dotnet/roslyn/issues/11502](https://github.com/dotnet/roslyn/issues/11502)?